src/mon/OSDMonitor.cc

```diff
@@ -2504,6 +2504,12 @@ bool OSDMonitor::can_mark_out(int i)
     return false;
   }
 
+  if (osdmap.is_admin_down(i)) {
```

Comment:
An alternate approach would be:

- ``osd admin-down`` command sets both ADMIN_DOWN and NOOUT flags
- if an osd starts with ADMIN_DOWN set, we clear the NOOUT flag

or, we could also a CLEAR_NOOUT_ON_START flag to and set that too.

I'm imagining a secondary goal here would be a simple ``ceph osd stop <osd.nnn>`` that sets the ADMIN_DOWN command but not the noout flag(s).  (And we could name it just CEPH_OSD_STOP.)  That would be more flags, but be more flexible.  What do you think?

Reply:
> I'm imagining a secondary goal here would be a simple `ceph osd stop <osd.nnn>` that sets the ADMIN_DOWN command but not the noout flag(s).

@liewegas Sounds better to me. Updated!